ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/location/CaptureLocationTaskMapFragment.kt

```diff
@@ -22,19 +22,33 @@ import android.view.ViewGroup
 import androidx.lifecycle.lifecycleScope
 import com.google.android.ground.ui.common.AbstractMapFragmentWithControls
 import com.google.android.ground.ui.common.BaseMapViewModel
+import com.google.android.ground.ui.datacollection.DataCollectionFragment
 import com.google.android.ground.ui.map.MapFragment
 import dagger.hilt.android.AndroidEntryPoint
 import kotlinx.coroutines.launch
+import timber.log.Timber
 
 @AndroidEntryPoint
-class CaptureLocationTaskMapFragment(private val viewModel: CaptureLocationTaskViewModel) :
-  AbstractMapFragmentWithControls() {
+class CaptureLocationTaskMapFragment : AbstractMapFragmentWithControls() {
 
   private lateinit var mapViewModel: CaptureLocationTaskMapViewModel
+  private lateinit var viewModel: CaptureLocationTaskViewModel
 
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
     mapViewModel = getViewModel(CaptureLocationTaskMapViewModel::class.java)
+    arguments?.let {
+      try {
+        val taskId = it.getString("taskId")
+        taskId?.let {
+          viewModel =
+            (requireParentFragment() as DataCollectionFragment).viewModel.getTaskViewModel(taskId)
+              as CaptureLocationTaskViewModel
+        }
+      } catch (e: Exception) {
+        Timber.e("CaptureLocationTaskMapFragment - $e")
+      }
```

Comment:
Please cleanup the debug logs.

Reply:
We do need this, so that we can get in production, right?

Reply:
Maybe also add the new method, `FirebaseCrashLogger().logException(t)` here if it's safe to do so, and in the other log areas?

And one more thing, by catching the error without rethrowing it, the app will be in a weird state where no `viewModel` is set and we skip everything that is in the `isInitialized` check. But these are critical for the functionality of the app, like updating map pins or capturing the location.

My question is, if the app finds itself in an impossible to recover state like this, should we log the exception and try to continue, or should we go back to the home screen and show and error to the user "An unexpected error has occurred. Please try again or contact your survey organizer"?  I'm not sure how to do that off the top of my head, so maybe just navigate to the home screen after logging. Also true for below.

Reply:
It's better that the app crashes than remains in an invalid state. But if we know that that can happen, we should try to fix the root cause rather than eating the exception.

Reply:
Also, @anandwana001 messages logged with `Timber.e()` aren't captured anywhere in prod. If you need to communicate critical debug info via Crashlytics, the method @sufyanAbbasi mentioned should work.

Reply:
Isn't it a regular error handling where we pass arguments to a new fragment and use it?
Not in a weird app state because this is something required to do. 
yes, we can use `FirebaseCrashLogger().logException(t)`

Reply:
> It's better that the app crashes than remains in an invalid state. But if we know that that can happen, we should try to fix the root cause rather than eating the exception.

This is more of a workaround we can do and user can just do a back and forth and keep continuing the data collection, no more crashes.

Later, as a core solution, we can work on the abstract layer and the ViewModel and support the orientation change to work things better
@gino-m

Reply:
> Isn't it a regular error handling where we pass arguments to a new fragment and use it?

Not sure what you mean about passing arguments to a new fragment is a regular error handling.. Can you rephrase?

> Not in a weird app state because this is something required to do.

It's generally considered bad practice to catch `Throwable` without managing potential consequences. The app will be in an invalid state if the view model couldn't be set since the view requires it to function, no?